dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceServiceTest.java

```diff
@@ -335,18 +340,195 @@ void testTrackedEntityInstanceGridWithNoDisplayAttributes()
 
         TrackedEntityInstanceQueryParams params = new TrackedEntityInstanceQueryParams();
         params.setOrganisationUnits( Sets.newHashSet( organisationUnit ) );
-        params.setTrackedEntityType( trackedEntityTypeA );
+        params.setTrackedEntityType( trackedEntityType );
 
         params.setQuery( new QueryFilter( QueryOperator.LIKE, ATTRIBUTE_VALUE ) );
 
         assertThrows( IllegalArgumentException.class,
             () -> entityInstanceService.getTrackedEntityInstancesGrid( params ) );
     }
 
+    @Test
+    void shouldOrderEntitiesByIdWhenParamCreatedAtSupplied()
+    {
+        injectSecurityContext( superUser );
+
+        entityInstanceA1.setCreated( DateTime.now().plusDays( 1 ).toDate() );
+        entityInstanceB1.setCreated( DateTime.now().toDate() );
+        entityInstanceC1.setCreated( DateTime.now().minusDays( 1 ).toDate() );
+        entityInstanceD1.setCreated( DateTime.now().plusDays( 2 ).toDate() );
+        addEntityInstances();
+
+        TrackedEntityInstanceQueryParams params = new TrackedEntityInstanceQueryParams();
+        params.setOrganisationUnits( Sets.newHashSet( organisationUnit ) );
+        params.setOrders( List.of( new OrderParam( "createdAt", OrderParam.SortDirection.ASC ) ) );
+
+        List<Long> teiIdList = entityInstanceService.getTrackedEntityInstanceIds( params, true, true );
+
+        Assertions.assertEquals( 4, teiIdList.size() );
+        Assertions.assertEquals( entityInstanceA1.getId(), teiIdList.get( 0 ) );
```

Comment:
Please replace your assertions with (so remove asserting on size as well)

`assertEquals(List.of(entityInstanceA1.getId(), entityInstanceB1.getId(), ...), teiIdList);`

as order matters. In cases order does not matter use https://github.com/dhis2/dhis2-core/blob/221cce4b8e9154e50c9a6458b713ba405cf27111/dhis-2/dhis-support/dhis-support-test/src/main/java/org/hisp/dhis/utils/Assertions.java#L67

This asserts the size is 4 and the ids are in the expected order. If the assertion is not met we will get an error message with the context needed to figure out whats wrong. The way you have written it it will fail with

```
org.opentest4j.AssertionFailedError: 
Expected :1
Actual   :3
```

which is not helpful when this test only fails on GitHub/Jenkins.